src/adagrams.js

```diff
@@ -1,15 +1,123 @@
+import _ from "underscore";
+
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const LETTER_SCORE = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const lettersArray = [];
+  let dictIndex = 0;
+// Use filter to compare if the times of the letter that has been pushed to lettersArray === its value   
+  while (
+    lettersArray.filter((x) => x === Object.keys(LETTER_POOL)[dictIndex])
+      .length <= Object.values(LETTER_POOL)[dictIndex]
+  ) {
+    lettersArray.push(Object.keys(LETTER_POOL)[dictIndex]);
+    dictIndex++;
+  }
+// draw 10 random letters from lettersArray using underscore.js 
+  return _.sample(lettersArray, 10);
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  // make a copy of lettersInHand array for removing elements 
+  const lettersInHandCopy = JSON.parse(JSON.stringify(lettersInHand));
+  let uppercasePool = lettersInHandCopy.map((letter) => letter.toUpperCase());
```

Comment:
Line 78 is not needed.  `.map` returns a new array, so it will create a capitalized version of the array.